README.md

```diff
@@ -1,1105 +1,20 @@
-somewhere it should be impossible to reach.   Even so, it is fine to log /
-exitall instead.
-
-
-Use string methods instead of the string module.
-
-String methods are always much faster and share the same API with
-unicode strings.  Override this rule if backward compatibility with
-Pythons older than 2.0 is required.
-
-Use `''.startswith()` and `''.endswith()` instead of string slicing to check
-for prefixes or suffixes.
-
-`startswith()` and `endswith()` are cleaner and less error prone.  For
-example:
-
-  Yes: 
-
-```python
-if foo.startswith('bar'):
-```
-
-  No:  
-
-```python
-if foo[:3] == 'bar':
-```
-
-Object type comparisons should always use `isinstance()` instead
-of comparing types directly.
-
-  Yes: 
-
-```python
-if isinstance(obj, int):
-```
-
-  No:  
-
-```python
-if type(obj) is type(1):
-```
-
-When checking if an object is a string, keep in mind that it might be a
-unicode string too!  In Python 2.3, `str` and `unicode` have a common base
-class, `basestring`, so you can do:
-
-```python
-if isinstance(obj, basestring):
-```
-
-The types module has the `StringTypes` type defined for that purpose, e.g.:
-
-```python
-import types
-if isinstance(obj, types.StringTypes):
-```
-
-For sequences, (strings, lists, tuples), use the fact that empty
-sequences are false.
-
-  Yes: 
-
-```python
-if not seq:
-if seq:
-```
-
-  No: 
-
-```python
-if len(seq)
-if not len(seq)
-```
-
-Don't write string literals that rely on significant trailing whitespace.  Such
-trailing whitespace is visually indistinguishable and some editors (or more
-recently, reindent.py) will trim them.
-
-Don't compare boolean values to `True` or `False` using `==`
-
-Yes:   
-
-```python
-if greeting:
-```
-
-No:    
-
-```python
-if greeting == True:
-```
-
-Worse: 
-
-```python
-if greeting is True:
-```
-
-However, there are a few cases where a function may return `True,` `False`, or
-other values.   In these cases, checking if a value with an unknown type is
-`True` or `False`, is allowed.
-
-Use the `%` string formatting operator only when necessary
-
-The string formatting operator `%` provides a convenient way to do `printf`-like
-substitution of variables in strings.  It's better in many cases to put
-variables in line when you don't need the more advanced options (like fixed
-length fields).  The issue is that the string formatting operator makes
-understanding the output difficult in many cases.   For example in the
-following example, it's difficult to read the code and determine what the
-resulting output will look like:
-
-```python
-print "python %s %s %s/vesselinfodir/ %s/ > /tmp/carter.out 2> /tmp/carter.err"%(carter_script, dist_char, prefix,prefix)
-```
-
-A better way to write this:
-
-```python
-print "python "+carter_script+" "+dist_char+" "+prefix+"/vesselinfodir/ "+prefix+"/ > /tmp/carter.out 2> /tmp/carter.err"
-```
-
-
-Use `else` statements for handling errors, not for normal flow in most cases.
```

Comment:
I am a bit unsure about this recommendation and whether we should keep it (it is marked for removal in this PR). 

IMO the example (below) is not about style but about correct programming. Although one may argue that the recommendation helps to avoid such programming errors? 

If we want to keep the recommendation I would make it about `switch` (i.e. if/elif/.../else statements in Python) only. Because if you only have if/else, I'd rather recommend the ["return early"](https://github.com/secure-systems-lab/code-style-guidelines/pull/21/files#diff-a2f3a3679195de40e4b40436591ed15733a553ae41afd6d0f4453b914cd71ddaR185-R186) pattern.